### DISCUSSION TITLE
Create an asynchronous logger but only for some trace levels

### DISCUSSION THREAD

**Discussion by skeleton18** [gabime/spdlog#2186] [Q&A] [answered]
Hello,

My project has an old (and not really working) logger system and I am looking for a COTS that fulfills my requirements.

Right now, I have several levels of tracing (DEBUG, INFO, WARNING, CRITICAL).
Is it possible to define an asynchronous logger that will queue DEBUG and INFO traces, but that will instantly print the WARNING and CRITICAL ones ?

Thanks for the help.

**Comment by tt4g** [comment, marked as answer]
There is no implementation that delays the output of a particular log level, but the OS buffers the output to console or file, and the output is delayed until the spdlog flush operation is invoked.

You can set the log level to force the output of buffered log messages with `spdlog::flush_on()` (or `spdlog::logger::flush_on()`).

See Wiki: https://github.com/gabime/spdlog/wiki/7.-Flush-policy

If you want to do queuing in your application instead of relying on OS buffering, you can create a custom sink (https://github.com/gabime/spdlog/wiki/4.-Sinks#implementing-your-own-sink).

**Reply by skeleton18** [reply]
Hello @tt4g,
Sorry for the late reply, personal issues. After reading your answer and doing some tests, it seems that spdlog matches my requirements. I do not need to implement queueing, just only an other sink to manage psql logging.
Thanks:
